lib/puppet/functions/mysql/password.rb

```diff
@@ -22,7 +22,7 @@
   def password(password, sensitive = false)
     password = password.unwrap if password.is_a?(Puppet::Pops::Types::PSensitiveType::Sensitive)
 
-    result_string = if %r{\*[A-F0-9]{40}$}.match?(password)
+    result_string = if %r{\*[A-F0-9]{40}$}.match?(password) || %r{0x24412430303524[A-F0-9]{63}$}.match?(password)
```

Comment:
Would be great to get the explanation from https://github.com/puppetlabs/puppetlabs-mysql/pull/1612#issuecomment-2032406107 in here to make it more discoverable for future code archaeologists. Even better would be a link to the documented format, if someone happens to know it!  😁

```suggestion
    # This magic string is the hex encoded form of `$A$005${SALT}{SHA DIGEST}`, matching MySQL's expected format
    result_string = if %r{\*[A-F0-9]{40}$}.match?(password) || %r{0x24412430303524[A-F0-9]{63}$}.match?(password)
```

Reply:
Hey thanks for the suggestion.
The only thing I can find is this blog https://dev.mysql.com/blog-archive/mysql-8-0-4-new-default-authentication-plugin-caching_sha2_password/ and https://crypto.stackexchange.com/questions/77427/whats-the-algorithm-behind-mysqls-sha256-password-hashing-scheme

Format: DELIMITER[digest_type]DELIMITER[iterations]DELIMITER[salt][digest]

DELIMITIER: $
digest_type: A => SHA256
iterations: 005 => 5*ITERATION_MULTIPLIER
salt: Random string. Length SALT_LENGTH
digest: SHA2 digest. Length STORED_SHA256_DIGEST_LENGTH